### PR TITLE
Some fixes

### LINKED ISSUE #451: traefik.toml constraints error: `Expected map but found 'string'.`
Using the `containous/traefik:v1.0.0-rc2` docker image against a consulCatalog backend:

traefik.toml snippet:
>[consulCatalog]
>  constraints = ["tag==foo"]

Errors with:
`2016/06/10 15:20:16 Error reading TOML config file /etc/traefik/traefik.toml : Type mismatch for 'main.TraefikConfiguration.ConsulCatalog': Type mismatch for 'provider.ConsulCatalog.Constraints': Type mismatch for types.Constraint. Expected map but found 'string'.`

Using the following command line option works:
`--consulCatalog.constraints=tag==foo`

### PR DESCRIPTION
**Pull request by emilevauge**
- [x] Fix constraints and add doc
- [x] Add Katacoda in docs
- [x] Move jobs backoff back into traefik https://github.com/cenk/backoff/pull/27#issuecomment-245842725
- [x] Clean web authentication

Fixes #451

### CODE REVIEW

**Comment by emilevauge** [comment]
ping @containous/traefik

File: backoff/job.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+const (
+	defaultMinJobInterval = 30 * time.Second
+)
+
+// JobBackOff is an exponential backoff implementation for long running jobs.
+// In long running jobs, an operation() that fails after a long Duration should not increments the backoff period.
+// If operation() takes more than MinJobInterval, Reset() is called in NextBackOff().
+type JobBackOff struct {
```
Comment (vdemeester) [inline backoff/job.go]: Hum I'm not sure about the naming of the package (and also the import we to). Shouldn't the package be `job` (or something like that) and this struct `BackOff` as an job implem' using a backoff strategie ?
And thus, the import wouldn't need to be aliased below.

Comment (emilevauge) [inline backoff/job.go]: Hum, import alias would still be needed ẁith `backoff.RetryNotify` no?
```
err := backoff.RetryNotify(operation, job.NewJobBackOff(backoff.NewExponentialBackOff()), notify)
```

Comment (vdemeester) [inline backoff/job.go]: no because you're package would be named `job`

Comment (vdemeester) [inline backoff/job.go]: (instead of `backoff` currently — which is why you need to do an alias as of this PR)

Comment (emilevauge) [inline backoff/job.go]: Indeed, fixed :)

**Comment by Russell-IO** [comment]
LGTM

**Comment by errm** [comment]
LGTM on rebase

**Pull request merged**